### PR TITLE
Add 7th challenge

### PR DESCRIPTION
**Pull request by arbazkiraak**

### CODE REVIEW

File: contracts/vulnerable/RareNFT.sol [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            tokenInfo[id] = Token({owner: msg.sender, value: nftPrice, rare: false});
+        }
+        minted[msg.sender] = true;
+        emit Minted(id, msg.sender);
+    }
+
+    function collect(uint256 id) external payable nonReentrant {
+        require(!collected[msg.sender], "RareNFT: you have already collected");
+        Token memory tk = tokenInfo[id];
+        require(tk.owner == msg.sender, "RareNFT: id doesn't belongs to you");
```
Comment (duncancmt) [inline contracts/vulnerable/RareNFT.sol:83]: Shouldn't you get owner information from the MockERC721?

File: contracts/vulnerable/RareNFT.sol [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        uint256 random = uint256(randHash) % _drawNum;
+        nonce++;
+        return random;
+    }
+
+    function mint(uint256 drawNum) external payable nonReentrant {
+        require(!minted[msg.sender], "RareNFT: you have already minted");
+        require(msg.value == nftPrice, "RareNFT: requires mint amount");
+        uint256 id = nftContract.mint();
+        uint256 randVal = _randGenerator(drawNum);
```
Comment (duncancmt) [inline contracts/vulnerable/RareNFT.sol:70]: Doesn't using `drawNum` as the modulus for the RNG make this _even more_ manipulable (especially when `luckyVal` is low)?

Comment (arbazkiraak) [reply contracts/vulnerable/RareNFT.sol:70]: This was intentional and on purpose to make it more manipulable.

File: contracts/vulnerable/RareNFT.sol (hunk 1 above)
Comment (arbazkiraak) [reply contracts/vulnerable/RareNFT.sol:83]: You are right, I should be querying it directly from the `nftContract` itself.

```
require(nftContract.ownerOf(id) == msg.sender,"RareNFT: id doesn't belongs to you");
```

Comment (arbazkiraak) [reply contracts/vulnerable/RareNFT.sol:83]: Fixed

File: contracts/vulnerable/RareNFT.sol (hunk 2 above)
Comment (arbazkiraak) [reply contracts/vulnerable/RareNFT.sol:70]: Removed the constructor argument and randomly sets the luckyVal. 

https://github.com/immunefi-team/community-challenges/pull/9/commits/d0fdd75a9312d70c2c292643e1d383571affc127

**Review by duncancmt — APPROVED** [review]

**Pull request merged**